src/DotNetLightning.Infrastructure/ChannelActor.fs

```diff
@@ -30,16 +30,20 @@ type ChannelActor(nodeParams: IOptions<NodeParams>,
     
     override this.HandleError (b: RBad) = unitTask {
             match b with
-            | RBad.Exception(ChannelException(ChannelError.Close(msg))) ->
-                let closeCMD =
-                    let spk = _nodeParams.ShutdownScriptPubKey |> Option.defaultValue (keysRepository.GetShutdownPubKey().WitHash.ScriptPubKey)
-                    ChannelCommand.Close({ CMDClose.ScriptPubKey = Some spk })
-                log.LogError(sprintf "Closing a channel for a node (%A) due to a following error. \n %s" (this.Channel.RemoteNodeId) msg)
-                log.LogError(sprintf "%s" msg)
-                return! (this :> IActor<_>).Put(closeCMD)
-            | RBad.Exception(ChannelException(ChannelError.Ignore(msg))) ->
-                log.LogWarning("Observed a following error in a channel. But ignoring")
-                log.LogWarning(msg)
+            | RBad.Exception(exp) ->
+                match exp with
+                | :? CloseChannelException as closeExp ->
+                    let closeCMD =
+                        let spk = _nodeParams.ShutdownScriptPubKey |> Option.defaultValue (keysRepository.GetShutdownPubKey().WitHash.ScriptPubKey)
+                        ChannelCommand.Close({ CMDClose.ScriptPubKey = Some spk })
+                    log.LogError(sprintf "Closing a channel for a node (%A) due to a following error:" (this.Channel.RemoteNodeId))
+                    log.LogError(sprintf "Exception message: %s" closeExp.Message)
+                    log.LogError(sprintf "Stack: %s" closeExp.StackTrace)
```

Comment:
in .NET, printing `ex.ToString()` prints both the message and the stacktrace (and the exception type), please use this instead of the two lines above

Reply:
My issue with `ToString` is that it doesn't have any words between those three components, only punctuation and whitespace. Since some of the errors in DotNetLightning span multiple lines, it becomes hard to see the structure. By using words it is easy to grep for e.g. `Stack:` and you will find this place.

Reply:
note that ToString() also walks the innerException tree (see some of my past nightmare stories, like this one: https://github.com/lefthandedgoat/canopy/commit/e8fea28a8771d8b38c14a32ec05cc76d3907ded8 )